apps/chat/src/components/Marketplace/TabRenderer.tsx

```diff
@@ -0,0 +1,225 @@
+    action: PublishActions;
+  }>();
+  const [detailsModel, setDetailsModel] = useState<DialAIEntityModel>();
+
+  const handleAddApplication = useCallback(() => {
+    setApplicationModel({
+      action: ApplicationActionType.ADD,
+    });
+  }, []);
+
+  const handleEditApplication = useCallback(
+    (entity: DialAIEntityModel) => {
+      dispatch(ApplicationActions.get(entity.id));
+      setApplicationModel({
+        entity,
+        action: ApplicationActionType.EDIT,
+      });
+    },
+    [dispatch],
+  );
+
+  const handleDeleteClose = useCallback(
+    (confirm: boolean) => {
+      if (confirm && deleteModel) {
+        if (deleteModel.action === DeleteType.REMOVE) {
+          const filteredModels = installedModels.filter(
+            (model) => deleteModel.entity.id !== model.id,
+          );
+          dispatch(ModelsActions.updateInstalledModels(filteredModels));
+        }
+        if (deleteModel.action === DeleteType.DELETE) {
+          dispatch(ApplicationActions.delete(deleteModel.entity));
+        }
+      }
+      setDeleteModel(undefined);
+    },
+    [deleteModel, installedModels, dispatch],
+  );
+
+  const handleSetPublishEntity = useCallback(
+    (entity: DialAIEntityModel, action: PublishActions) =>
+      setPublishModel({
+        entity: {
+          name: entity.name,
+          id: ApiUtils.decodeApiUrl(entity.id),
+          folderId: getFolderIdFromEntityId(entity.id),
+        },
+        action,
+      }),
+    [],
+  );
+
+  const handlePublishClose = useCallback(() => setPublishModel(undefined), []);
+
+  const handleDelete = useCallback(
+    (entity: DialAIEntityModel) => {
+      setDeleteModel({ entity, action: DeleteType.DELETE });
+    },
+    [setDeleteModel],
+  );
+
+  const handleRemove = useCallback(
+    (entity: DialAIEntityModel) => {
+      setDeleteModel({ entity, action: DeleteType.REMOVE });
+    },
+    [setDeleteModel],
+  );
+
+  const handleCardClick = useCallback(
+    (entity: DialAIEntityModel) => {
+      setDetailsModel(entity);
+    },
+    [setDetailsModel],
+  );
+
+  const handleCloseApplicationDialog = useCallback(
+    () => setApplicationModel(undefined),
+    [setApplicationModel],
+  );
+
+  const handleCloseDetailsDialog = useCallback(
+    () => setDetailsModel(undefined),
+    [setDetailsModel],
+  );
+
+  const filteredModels = useMemo(() => {
+    if (selectedTab === MarketplaceTabs.MY_APPLICATIONS) {
+      return entities.filter(
+        (entity) => !!installedModels.find((model) => model.id === entity.id),
+      );
+    }
+    return entities;
+  }, [selectedTab, entities, installedModels]);
+
+  return (
+    <>
+      <header className="mb-4">
+        <MarketplaceBanner />
+        <SearchHeader
+          items={filteredModels.length}
+          onAddApplication={handleAddApplication}
+        />
+      </header>
+
+      <CardsList
+        title={
+          selectedTab === MarketplaceTabs.HOME ? 'All applications' : undefined
```

Comment:
and what about translation here?

Reply:
translation is applied inside the CardList